src/index.js

```diff
@@ -35,37 +39,59 @@ export default async function educandu(options) {
     process.exit(hasError ? 1 : 0);
   });
 
-  process.on('uncaughtException', err => {
-    logger.fatal(err);
-    Graceful.exit(1);
-  });
-
   try {
-    logger.info('Starting application');
-
     container = await bootstrapper.createContainer(options);
+    const maintenanceService = container.get(MaintenanceService);
     const educanduServer = container.get(EducanduServer);
     const serverConfig = container.get(ServerConfig);
+    const userService = container.get(UserService);
+    const database = container.get(Database);
 
-    logger.info('Starting server');
-    educanduServer.listen((err, port) => {
-      if (err) {
-        logger.fatal(err);
-        Graceful.exit(1);
+    logger.info('Starting application');
+
+    logger.info('Starting server in maintenance mode');
+    const port = await educanduServer.listen({ maintenance: true });
+    logger.info(`Server listening on http://localhost:${port}`);
+
+    if (options.skipMongoMigrations) {
+      logger.info('Skipping database migrations');
+    } else {
+      logger.info('Starting database migrations');
+      await maintenanceService.runMaintenance();
```

Comment:
Since you opted for this extendable more generic approach, the log messages no longer fit.

How about `maintenanceService.runMaintenance({ skipMongoMigrations: options.skipMongoMigrations })` and having the service do the particular db related logging, and in the future any other maintenance actions?

Reply:
Sounds like a good idea. I will then also move the db checks into this maintenance service and keep the server config value!

Reply:
Looks good to me!